api/src/repositories/submission-repository.ts

```diff
@@ -398,35 +398,108 @@ export class SubmissionRepository extends BaseRepository {
    * @memberof SubmissionRepository
    */
   async getSpatialComponentCountByDatasetId(datasetId: string): Promise<ISpatialComponentCount[]> {
+    if (await this.isSystemUserAdmin()) {
+      return this._getSpatialComponentCountByDatasetIdAsAdmin(datasetId);
+    }
+
+    return this._getSpatialComponentCountByDatasetIdWithSecurity(datasetId);
+  }
+
+  /**
+   * Get spatial component counts by dataset id for admins
+   *
+   * @param {string} datasetId
+   * @return {*}  {Promise<ISpatialComponentCount[]>}
+   * @memberof SubmissionRepository
+   */
+  async _getSpatialComponentCountByDatasetIdAsAdmin(datasetId: string): Promise<ISpatialComponentCount[]> {
     const sqlStatement = SQL`
-      WITH with_submission as (
         SELECT
-          submission.submission_id
-        FROM
-          submission
-        WHERE
-          submission.uuid = ${datasetId}
-        AND
-          submission.record_end_date is NULL
+          features_array #> '{properties, type}' spatial_type,
+          count(features_array #> '{properties, type}')::integer count
+        FROM 
+          submission_spatial_component ssc,
+          jsonb_array_elements(ssc.spatial_component -> 'features') features_array,
+          submission s
+        WHERE s.uuid = ${datasetId}
+        AND ssc.submission_id = s.submission_id
+        AND s.record_end_date is null
+        GROUP BY spatial_type;
+      `;
+    const response = await this.connection.sql<ISpatialComponentCount>(sqlStatement);
+    return response.rows;
+  }
+
+  /**
+   * Get spatial component counts by dataset id applying security rules
+   *
+   * @param {string} datasetId
+   * @return {*}  {Promise<ISpatialComponentCount[]>}
+   * @memberof SubmissionRepository
+   */
+  async _getSpatialComponentCountByDatasetIdWithSecurity(datasetId: string): Promise<ISpatialComponentCount[]> {
+    const knex = getKnex();
+    const queryBuilder = knex
+      // get security transforms
+      .with('with_user_security_transform_exceptions', (qb) => {
+        qb.select(knex.raw('array_agg(suse.security_transform_id) as user_security_transform_exceptions'))
+          .from('system_user_security_exception as suse')
+          .where('suse.system_user_id', this.connection.systemUserId());
+      })
+      // filter spatial components for data set
+      .with(
+        'with_filtered_spatial_component_with_security_transforms',
+        knex.raw(`
+      SELECT 
```

Comment:
nit: can you indent the body of this knex.raw (and the 2 other instances below) a bit.

Reply:
having to manually format these template strings is a bit annoying, but helps so much in legibility

Reply:
done